src/main/java/codeu/controller/ConversationServlet.java

```diff
@@ -101,23 +115,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
 
     String conversationTitle = request.getParameter("conversationTitle");
     if (!conversationTitle.matches("[\\w*]*")) {
-      request.setAttribute("error", "Please enter only letters and numbers.");
-      request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
-      return;
+      if (conversationTitle.contains("#")) {
+        String hashtagTitle = (String)conversationTitle.substring(conversationTitle.indexOf("#") + 1);
+        if (hashtagStore.isTitleTaken(hashtagTitle) == false) {
+          Hashtag hashtag = new Hashtag(UUID.randomUUID(), user.getId(), hashtagTitle, Instant.now());
+        }
+      } else {
+        request.setAttribute("error", "Please enter only letters and numbers.");
+        request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
+        return;
+      }
     }
 
     if (conversationStore.isTitleTaken(conversationTitle)) {
       // conversation title is already taken, just go into that conversation instead of creating a
       // new one
-      response.sendRedirect("/chat/" + conversationTitle);
+      if (conversationTitle.contains("#")) {
+        String hashtagTitle = (String)conversationTitle.substring(conversationTitle.indexOf("#") + 1);
+        response.sendRedirect("/chat/" + hashtagTitle);
+      } else {
+        response.sendRedirect("/chat/" + conversationTitle);
+      }
       return;
     }
 
-    Conversation conversation =
-        new Conversation(UUID.randomUUID(), user.getId(), conversationTitle, Instant.now());
+    Conversation conversation;
+    if (conversationTitle.contains("#")) {
+      conversation = new Conversation(UUID.randomUUID(), user.getId(), conversationTitle.substring(conversationTitle.indexOf("#") + 1), Instant.now());
+    } else {
+      conversation = new Conversation(UUID.randomUUID(), user.getId(), conversationTitle, Instant.now());
+    }
 
     conversationStore.addConversation(conversation);
     System.out.println(conversationTitle);
-    response.sendRedirect("/chat/" + conversationTitle);
+    if (conversationTitle.contains("#")) {
+      String hashtagTitle = (String)conversationTitle.substring(conversationTitle.indexOf("#") + 1);
+      response.sendRedirect("/chat/" + hashtagTitle);
```

Comment:
this might not be what you want to happen

the way this code is set up, i can create 2 different conversations for #foo and foo (since we're using conversation title to check for uniqueness, and conversation title contains the pound sign). However, they all redirect to the same /chat/foo page which implies that they are the same conversation... Please reconcile...

I HIGHLY suggest we go with Jorge's suggestion during the working session to keep hashtag recognition out of the conversation title and have a separate textbox with list of tags. however since time is tight, your other options are to either make every single conversation title a hashtag entity as well regardless of whether hashtag # is present or we don't support hashtagging conversations (or if you have other ideas on how to reconcile #foo with foo)